packages/app/src/cli/services/context/prompts.ts

```diff
@@ -66,40 +107,79 @@ export async function deployConfirmationPrompt(
   }
 
   if (onlyRemote.length > 0) {
-    let missingLocallySection: InfoTableSection = {
-      header: 'Missing locally',
-      items: onlyRemote.map((source) => source.title),
-    }
+    infoTable.push({header: 'Missing locally', items: onlyRemote.map((source) => source.title)})
+  }
 
-    if (deploymentMode === 'unified' || deploymentMode === 'unified-skip-release') {
-      missingLocallySection = {
-        ...missingLocallySection,
-        header: 'Removed',
-        color: 'red',
-        helperText: 'Will be removed for users when this version is released.',
-      }
-    }
+  return infoTable
+}
 
-    infoTable.push(missingLocallySection)
+async function buildUnifiedDeploymentInfoPrompt(
+  apiKey: string,
+  token: string,
+  localRegistration: IdentifiersExtensions,
+  toCreate: LocalSource[],
+  dashboardOnly: RemoteSource[],
+  deploymentMode: DeploymentMode,
+) {
+  if (deploymentMode === 'legacy') return []
+
+  const activeAppVersion = await fetchActiveAppVersion({token, apiKey})
+
+  if (!activeAppVersion.app.activeAppVersion) return []
+
+  const infoTable: InfoTableSection[] = []
+
+  const nonDashboardActiveAppRegistrations = activeAppVersion.app.activeAppVersion.appModuleVersions.filter(
+    (module) => module.specification.options.managementExperience !== 'dashboard',
+  )
+
+  const toCreateFinal = [
+    ...new Set(
+      Object.entries(localRegistration)
+        .filter(
+          (validLocalRegistration) =>
+            !nonDashboardActiveAppRegistrations
+              .map((remoteRegistration) => remoteRegistration.registrationUuid)
+              .includes(validLocalRegistration[1]),
+        )
+        .map((source) => source[0])
+        .concat(toCreate.map((source) => source.localIdentifier)),
+    ),
+  ]
```

Comment:
This took me a while to read. What about a more verbose approach?: 
```suggestion
const localObjects = Object.entries(localRegistration)
const remoteUuids = nonDashboardActiveAppRegistrations.map((regis) => regis.registrationUuid)
const localUuidsToCreate = localObjects.filter((local) => remoteUuids.includes(local[1])).map((local) => local[0])
const allToCreate = localUuidsToCreate.concat(toCreate.map((source) => source.localIdentifier))
const toCreateFinal = [...new Set(allToCreate)]
```
What do you think?

Reply:
yep, totally agree. First that "mapper" expression was really simple bit it ended up being difficult to read. I should have refactored it as your suggestion

Reply:
Something like this might work too, and is maybe clearer due to not using `[0]` and `[1]`:

```suggestion
  const registrationUuids =
    nonDashboardActiveAppRegistrations.map((remoteRegistration) => remoteRegistration.registrationUuid)
  const toCreateSet = new Set()
  for (const [identifier, uuid] of Object.entries(localRegistration)) {
    if (registrationUuids.includes(uuid)) toCreateSet.add(identifier)
  }
  const toCreateFinal = Array.from(toCreateSet)
```